cmds/exec_agent/sync.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package main
+
+import (
+	"bytes"
+	"sync"
+)
+
+// SafeSignal is a goroutine safe signalling mechanism
```

Comment:
ok, ill add some more text. Initially this was a chan and close(chan) but that doesnt work for multiple consumers because the 2nd close panics on a closed channel. The point here is this signal can be triggered multiple times and it's a noop for all but 1st call.